aptos-move/move-examples/fungible_asset/stablecoin/sources/fiat_token.move

```diff
@@ -0,0 +1,420 @@
+        let fiat_token_config = authorized_borrow_refs(minter, metadata_object);
+
+        // burn the tokens from the minter's primary store
+        let from_wallet = primary_fungible_store::primary_store(signer::address_of(minter), metadata_object);
+        fungible_asset::burn_from(&fiat_token_config.burn_ref, from_wallet, amount);
+
+        // TODO: emit burning event (??) or is FA events sufficient, indexing could be done that way. any other events?
+    }
+    
+    /// Transfer fiat tokens from `from` address to `to` address.
+    /// Authorizer can be any signer with sufficient balance in their primary store.
+    public entry fun transfer(from: &signer, to: address, amount: u64) acquires FiatTokenAccessInfo {
+        // SECURITY TODO: check if contract is paused
+
+        // check if `from` or  `to` address is blacklisted
+        assert!(!is_blacklisted(signer::address_of(from)), error::permission_denied(EBLACKLISTED_ADDRESS));
+        assert!(!is_blacklisted(to), error::permission_denied(EBLACKLISTED_ADDRESS));
+
+        // check if amount is greater than `from` address's balance
+        let from_addr = signer::address_of(from); 
+        assert!(balance_of_primary_store(from_addr) < amount, error::permission_denied(EINSUFFICIENT_BALANCE));
+
+        // transfer the tokens from `from` address to `to` address using transfer ref
+        // transfer() with calls withdraw() and deposit(), emitting two events respectively
+        let metadata_object = get_metadata_object();
+        let from_wallet = primary_fungible_store::primary_store(from_addr, metadata_object);
+        let to_wallet = primary_fungible_store::ensure_primary_store_exists(to, metadata_object);
+        fungible_asset::transfer(from, from_wallet, to_wallet, amount);
+
+        // TODO: do we need to emit an event here?
+    }
+    
+    /// Add a new minter with an allowance or update the minter's allowance.
+    /// Only callable by the master minter.
+    /// Returns true if operation is successful.
+    public(friend) fun configure_minter(master_minter: &signer, minter: address, allowance: u64): bool acquires MasterMinterConfig, FiatTokenAccessInfo {
+        // SECURITY TODO: check if contract is paused
+
+        // SECURITY TODO: check if caller is masterMinter
+        let caller_addr = signer::address_of(master_minter);
+        let master_minter_addr = borrow_global<MasterMinterConfig>(@circle).master_minter;
+        assert!(caller_addr == master_minter_addr, error::permission_denied(ENOT_MASTER_MINTER));
+
+        // SECURITY TODO: check if minter is blacklisted
+        assert!(!is_blacklisted(minter), error::permission_denied(EBLACKLISTED_ADDRESS));
+
+        // retrieve the access info and add a new minter and allowance OR update the minter's allowance if already exists
+        let access_info = borrow_global_mut<FiatTokenAccessInfo>(@circle);
+        smart_table::upsert(&mut access_info.minters, minter, allowance);
```

Comment:
Should note above that it will not increment the allowance, but replace it entirely.